tests/compilerapi/tcompilerapi.nim

```diff
@@ -28,27 +27,28 @@ proc main() =
     quit "script does not export a proc of the name: 'hostProgramRunsThis'"
   let res = intr.callRoutine(foreignProc, [newFloatNode(nkFloatLit, 0.9),
                                            newFloatNode(nkFloatLit, 0.1)])
-  if res.kind == nkFloatLit:
-    echo res.floatVal
-  else:
-    echo "bug!"
+  doAssert res.kind == nkFloatLit
+  echo res.floatVal
 
   let foreignValue = selectUniqueSymbol(intr, "hostProgramWantsThis")
   if foreignValue == nil:
     quit "script does not export a global of the name: hostProgramWantsThis"
   let val = intr.getGlobalValue(foreignValue)
-  if val.kind in {nkStrLit..nkTripleStrLit}:
-    echo val.strVal
-  else:
-    echo "bug!"
-
+  doAssert val.kind in {nkStrLit..nkTripleStrLit}
+  echo val.strVal
   destroyInterpreter(intr)
 
-if existsEnv("NIM_EXE_NOT_IN_PATH"):
-  # effectively disable this test as 'nim' is not in the PATH so tcompilerapi
-  # cannot find Nim's standard library:
-  echo "top level statements are executed!"
-  echo "2.0"
-  echo "my secret"
-else:
-  main()
+main()
+
+block issue9180:
+  proc evalString(code: string, moduleName = "script.nim"):auto=
+    let stream = llStreamOpen(code)
+    defer: llStreamClose(stream)
```

Comment:
Don't use ``defer``.

Reply:
done to move forward with this PR although I (and others) think defer is preferable, see https://forum.nim-lang.org/t/4022#25111